common/app/views/support/cleaner/AmpEmbedCleaner.scala

```diff
@@ -85,30 +92,86 @@ case class AmpEmbedCleaner(article: Article) extends HtmlCleaner {
     }
 
     def getTrackIdFromUrl(soundcloudUrl: String): Option[String] = {
-      val pattern = ".*soundcloud.com%2Ftracks%2F(\\d+).*".r
-      soundcloudUrl match {
-        case pattern(trackId) => Some(trackId)
+      val pattern = ".*api.soundcloud.com/tracks/(\\d+).*".r
+      URLDecoder.decode(soundcloudUrl,"UTF-8") match {
+        case pattern(trackId) => {
+          Some(trackId)}
         case _ => None
       }
     }
 
+    def getSoundCloudElement(document: Document, iframeElement: Element): Option[Element] = {
+      val trackId = AmpSoundcloud.getTrackIdFromUrl(iframeElement.attr("src"))
+      trackId.map(id => AmpSoundcloud.createElement(document, id))
+      /*  if (trackId.nonEmpty) {
+        Option(AmpSoundcloud.createElement(document, trackId.get))
+    } else {
+        None
+      }*/
+    }
+
+  }
+
+
+  //creates embeds for any elements of type "element-audio"
```

Comment:
Nit: I actually found this comments a bit confusing, because they are not all relating to `AmpAudioElements`. 

I think comments are great if they explain something that is confusing, but they aren't always needed. For example, if they name of the function/object is really clear and already explains what it is doing, you definitely don't need a comment. 

Likewise, comments should only be relevant. `//other embeds are created as amp-iframe` is only really relevant for 144 and below so this comment could be moved there if it is still useful.